### PR TITLE
Fix quick reference guide links

### PR DESCRIPTION
**Pull request by rpalcolea**
Fixes broken links between Quick Reference and the Guide.

Example:  http://docs.grails.org/latest/ref/Plug-ins/URL%20mappings.html in

> Refer to the section on URL Mapping in the Grails user guide which details Grails' URL mappings.

If you click "URL Mapping” it doesn’t take you to http://docs.grails.org/latest/guide/theWebLayer.html#urlmappings anymore

### CODE REVIEW

File: src/en/ref/Domain Classes/createCriteria.adoc [hunk 1]
```diff
@@ -84,51 +86,158 @@ c.list { ... }
 
 Below is a node reference for each criterion method:
 
-[cols="3*", options="header"]
+[format="csv", options="header"]
```
Comment (graemerocher) [inline src/en/ref/Domain Classes/createCriteria.adoc]: I think the changes to this table break it. Could you revert the changes here.

File: src/en/ref/Domain Classes/createCriteria.adoc [hunk 2]
```diff
@@ -145,17 +254,48 @@ def branchCount = c.get {
 
 The following table summarizes the different projections and what they do:
 
-[cols="3*", options="header"]
+[format="csv", options="header"]
```
Comment (graemerocher) [inline src/en/ref/Domain Classes/createCriteria.adoc]: Same with this table

File: src/en/ref/Domain Classes/errors.adoc [hunk 3]
```diff
@@ -30,9 +30,9 @@ else {
 === Description
 
 
-The `errors` property is used by Grails during link:theWebLayer.html#dataBinding[data binding] to store type conversion errors and during link:validation.html[validation] when calling the link:../Domain%20Classes/validate.html[validate] or link:../Domain%20Classes/save.html[save] methods.
+The `errors` property is used by Grails during link:{guidePath}/theWebLayer.html#dataBinding[data binding] to store type conversion errors and during link:{guidePath}validation.html[validation] when calling the link:validate.html[validate] or link:save.html[save] methods.
 
-You can also add your own errors using the {springapi}org/springframework/validation/Errors#reject(java/lang/String).html[reject] and {springapi}org/springframework/validation/Errors#rejectValue(java/lang/String,%20java/lang/String).html[rejectValue] methods:
+You can also add your own errors using the http://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/validation/Errors#reject(java/lang/String).html[reject] and http://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/validation/Errors#rejectValue(java/lang/String,%20java/lang/String).html[rejectValue] methods:
```
Comment (graemerocher) [inline src/en/ref/Domain Classes/errors.adoc]: Also these spring links don't need to be changed

File: src/en/ref/Plug-ins/controllers.adoc [hunk 4]
```diff
@@ -44,10 +44,10 @@ This plugin configures Grails to use Spring MVC at its core to deal with web req
 
 Configured Spring Beans:
 
-* `exceptionHandler` - An instance of {apiDocs}org/grails/web/errors/GrailsExceptionResolver.html[GrailsExceptionResolver] for dealing with exceptions
-* `multipartResolver` - An instance of {springapi}org/springframework/web/multipart/commons/CommonsMultipartResolver.html[CommonsMultipartResolver] used for dealing with file uploads using Apache Commons File Upload. If you do not wish to use this, set grails.disableCommonsMultipart to true in `application.groovy`. Be aware that disabling multipart handling will effect the behaviour of g:actionSubmit which needs to inspect the parameters (requiring the multipart request to be parsed) during the URL mapping phase.
+* `exceptionHandler` - An instance of http://docs.grails.org/latest/api/org/grails/web/errors/GrailsExceptionResolver.html[GrailsExceptionResolver] for dealing with exceptions
```
Comment (graemerocher) [inline src/en/ref/Plug-ins/controllers.adoc]: This change is also unnecessary

File: src/en/ref/Plug-ins/dataSource.adoc [hunk 5]
```diff
@@ -39,8 +39,8 @@ dataSource {
 === Description
 
 
-Refer to the section on <<dataSource,Data Sources>> in the Grails user guide.
+Refer to the section on link:{guidePath}/conf.html#dataSource[Data Sources] in the Grails user guide.
 
 Configured Spring Beans:
 
-* `dataSource` - An instance of the Apache DBCP library's `org.apache.commons.dbcp.BasicDataSource` class or a {springapi}org/springframework/jndi/JndiObjectFactoryBean.html[JndiObjectFactoryBean] in the case of a JNDI `DataSource`
+* `dataSource` - An instance of the Apache DBCP library's `org.apache.commons.dbcp.BasicDataSource` class or a http://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/jndi/JndiObjectFactoryBean.html[JndiObjectFactoryBean] in the case of a JNDI `DataSource`
```
Comment (graemerocher) [inline src/en/ref/Plug-ins/dataSource.adoc]: This change too is unnecessary

File: src/en/ref/Plug-ins/domainClasses.adoc [hunk 6]
```diff
@@ -27,11 +27,11 @@ class Book {
 === Description
 
 
-Refer to the section on link:../../guide/GORM.html[GORM] in the Grails user guide which details how to create Grails domain classes.
+Refer to the section on link:http://gorm.grails.org/6.0.x/hibernate/manual/[GORM] in the Grails user guide which details how to create Grails domain classes.
 
 Configured Spring Beans given a domain class of `Book`:
 
-* `BookValidator` - A {apiDocs}org/grails/validation/GrailsDomainClassValidator.html[GrailsDomainClassValidator] that validates a domain class' defined link:validation.html#constraints[Constraints].
+* `BookValidator` - A http://docs.grails.org/latest/api/org/grails/validation/GrailsDomainClassValidator.html[GrailsDomainClassValidator] that validates a domain class' defined link:validation.html#constraints[Constraints].
```
Comment (graemerocher) [inline src/en/ref/Plug-ins/domainClasses.adoc]: This change too is unnecessary

File: src/en/ref/Plug-ins/domainClasses.adoc [hunk 7]
```diff
@@ -27,11 +27,11 @@ class Book {
...
 * `BookPersistentClass` - The `Class` for the domain class `Book`
-* `BookDomainClass` - The {apiDocs}grails/core/GrailsDomainClass.html[GrailsDomainClass] instance which understands the conventions defined within a domain class and is used by link:../../guide/GORM.html[GORM] to perform ORM mapping.
+* `BookDomainClass` - The http://docs.grails.org/latest/api/grails/core/GrailsDomainClass.html[GrailsDomainClass] instance which understands the conventions defined within a domain class and is used by link:GORM.html[GORM] to perform ORM mapping.
```
Comment (graemerocher) [inline src/en/ref/Plug-ins/domainClasses.adoc]: This change too is unnecessary

File: src/en/ref/Plug-ins/hibernate.adoc [hunk 8]
```diff
@@ -27,14 +27,14 @@ class Book {
 === Description
 
 
-Refer to the section on link:../../guide/GORM.html[GORM] in the Grails user guide which details how to create Grails domain classes.
+Refer to the section on link:GORM.html[GORM] in the Grails user guide which details how to create Grails domain classes.
 
 Configured Spring Beans:
 
-* `dialectDetector` - An instance of {apiDocs}org/grails/orm/hibernate/support/HibernateDialectDetectorFactoryBean.html[HibernateDialectDetectorFactoryBean] that attempts to automatically detect the Hibernate http://docs.jboss.org/hibernate/orm/current/javadocs/org/hibernate/dialect/Dialect.html[Dialect] which is used to communicate with the database.
+* `dialectDetector` - An instance of http://docs.grails.org/latest/api/org/grails/orm/hibernate/support/HibernateDialectDetectorFactoryBean.html[HibernateDialectDetectorFactoryBean] that attempts to automatically detect the Hibernate http://docs.jboss.org/hibernate/orm/current/javadocs/org/hibernate/dialect/Dialect.html[Dialect] which is used to communicate with the database.
```
Comment (graemerocher) [inline src/en/ref/Plug-ins/hibernate.adoc]: This change too is unnecessary

File: src/en/ref/Plug-ins/hibernate.adoc [hunk 9]
```diff
@@ -27,14 +27,14 @@ class Book {
...
 * `hibernateProperties` - A `Map` of Hibernate properties passed to the `SessionFactory`
-* `lobHandlerDetector` - An instance of {apiDocs}org/grails/orm/hibernate/support/SpringLobHandlerDetectorFactoryBean.html[SpringLobHandlerDetectorFactoryBean] that attempts to automatically detect the Spring LobHandler to use based on the database.
+* `lobHandlerDetector` - An instance of http://docs.grails.org/latest/api/org/grails/orm/hibernate/support/SpringLobHandlerDetectorFactoryBean.html[SpringLobHandlerDetectorFactoryBean] that attempts to automatically detect the Spring LobHandler to use based on the database.
```
Comment (graemerocher) [inline src/en/ref/Plug-ins/hibernate.adoc]: This change too is unnecessary

File: src/en/ref/Plug-ins/hibernate.adoc [hunk 10]
```diff
@@ -27,14 +27,14 @@ class Book {
...
-* `openSessionInViewInterceptor` - An instance of {apiDocs}org/grails/orm/hibernate/support/GrailsOpenSessionInViewFilter.html[GrailsOpenSessionInViewFilter] that deals with Grails' http://docs.jboss.org/hibernate/orm/current/javadocs/org/hibernate/Session.html[Session] management.
+* `transactionManager` - An instance of Spring's http://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/orm/hibernate3/HibernateTransactionManager.html[HibernateTransactionManager] class
+* `persistenceInterceptor` - An instance of http://docs.grails.org/latest/api/org/grails/orm/hibernate/support/HibernatePersistenceContextInterceptor.html[HibernatePersistenceContextInterceptor] that abstracts away how persistence is used from the controller/view layer so that Grails can be used without GORM.
+* `openSessionInViewInterceptor` - An instance of http://docs.grails.org/latest/api/org/grails/orm/hibernate/support/GrailsOpenSessionInViewFilter.html[GrailsOpenSessionInViewFilter] that deals with Grails' http://docs.jboss.org/hibernate/orm/current/javadocs/org/hibernate/Session.html[Session] management.
```
Comment (graemerocher) [inline src/en/ref/Plug-ins/hibernate.adoc]: This change too is unnecessary

File: src/en/ref/Plug-ins/i18n.adoc [hunk 11]
```diff
@@ -27,10 +27,10 @@ default.not.found.message={0} not found with id {1}
...
-* `localeResolver` - An instance of the Spring {springapi}org/springframework/web/servlet/i18n/CookieLocaleResolver.html[CookieLocaleResolver] class that is used to store the user locale in a cookie.
+* `messageSource` - An instance the Spring http://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/context/support/ReloadableResourceBundleMessageSource.html[ReloadableResourceBundleMessageSource] class which is used to resolve messages.
+* `localeChangeInterceptor` - An instance of the Spring http://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/web/servlet/i18n/LocaleChangeInterceptor.html[LocaleChangeInterceptor] class which is used to switch locales using a `lang` request parameter.
+* `localeResolver` - An instance of the Spring http://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/web/servlet/i18n/CookieLocaleResolver.html[CookieLocaleResolver] class that is used to store the user locale in a cookie.
```
Comment (graemerocher) [inline src/en/ref/Plug-ins/i18n.adoc]: This change too is unnecessary

File: src/en/ref/Servlet API/request.adoc [hunk 12]
```diff
@@ -6,7 +6,7 @@
 === Purpose
 
 
-The `request` object is an instance of the Servlet API's {javaee}javax/servlet/http/HttpServletRequest.html[HttpServletRequest] interface
+The `request` object is an instance of the Servlet API's http://docs.oracle.com/javaee/1.4/api/javax/servlet/http/HttpServletRequest.html[HttpServletRequest] interface
```
Comment (graemerocher) [inline src/en/ref/Servlet API/request.adoc]: This change too is unnecessary

File: src/en/ref/Servlet API/request.adoc [hunk 13]
```diff
@@ -27,21 +27,21 @@ class BookController {
 === Description
 
 
-The {javaee}javax/servlet/http/HttpServletRequest.html[HttpServletRequest] class is useful for, amongst other things, obtaining request headers, storing request scoped attributes and establishing information about the client. Refer to the Servlet API's javadocs for further information.
+The http://docs.oracle.com/javaee/1.4/api/javax/servlet/http/HttpServletRequest.html[HttpServletRequest] class is useful for, amongst other things, obtaining request headers, storing request scoped attributes and establishing information about the client. Refer to the Servlet API's javadocs for further information.
```
Comment (graemerocher) [inline src/en/ref/Servlet API/request.adoc]: This change too is unnecessary

File: src/en/ref/Servlet API/request.adoc [hunk 14]
```diff
@@ -27,21 +27,21 @@ class BookController {
...
 
 Grails enhances the `HttpServletRequest` instance by adding the following new properties and methods:
 
 * `XML` - An instance of `XmlSlurper`'s http://groovy.codehaus.org/api/groovy/util/slurpersupport/GPathResult.html[GPathResult] class that allows parsing of an incoming XML request (useful for REST).
-* `JSON` - An instance of Grails' http://docs.grails.org/3.1.x/apiorg/codehaus/groovy/grails/web/json/JSONObject.html[JSONObject] class that allows parsing of an incoming JSON request (useful for JSON based REST).
+* `JSON` - An instance of Grails' http://docs.grails.org/3.2.x/apiorg/codehaus/groovy/grails/web/json/JSONObject.html[JSONObject] class that allows parsing of an incoming JSON request (useful for JSON based REST).
```
Comment (graemerocher) [inline src/en/ref/Servlet API/request.adoc]: This change too is unnecessary

File: src/en/ref/Servlet API/request.adoc [hunk 15]
```diff
@@ -61,7 +61,7 @@ def title = request.XML?.title
 render "The Title is $title"
 ----
 
-Request attributes which are normally accessible from the {javaee}javax/servlet/ServletRequest#getAttribute(java/lang/String).html[getAttribute] can also be indexed into using the array index operator or de-reference operator:
+Request attributes which are normally accessible from the http://docs.oracle.com/javaee/1.4/api/javax/servlet/ServletRequest#getAttribute(java/lang/String).html[getAttribute] can also be indexed into using the array index operator or de-reference operator:
```
Comment (graemerocher) [inline src/en/ref/Servlet API/request.adoc]: This change too is unnecessary

File: src/en/ref/Servlet API/response.adoc [hunk 16]
```diff
@@ -6,7 +6,7 @@
 === Purpose
 
 
-The `response` object is an instance of the Servlet API's {javaee}javax/servlet/http/HttpServletResponse.html[HttpServletResponse] class
+The `response` object is an instance of the Servlet API's http://docs.oracle.com/javaee/1.4/api/javax/servlet/http/HttpServletResponse.html[HttpServletResponse] class
```
Comment (graemerocher) [inline src/en/ref/Servlet API/response.adoc]: This change too is unnecessary

File: src/en/ref/Servlet API/response.adoc [hunk 17]
```diff
@@ -26,10 +26,10 @@ class BookController {
 === Description
 
 
-The Servlet API's `HttpServletResponse` class can be used within Grails to perform all typical activities such as writing out binary data, writing directly to the response and sending error response codes to name but a few. Refer to the documentation on the {javaee}javax/servlet/http/HttpServletResponse.html[HttpServletResponse] class in the Servlet API for more information. Grails provides the following additional methods and/or properties on the `response` object
+The Servlet API's `HttpServletResponse` class can be used within Grails to perform all typical activities such as writing out binary data, writing directly to the response and sending error response codes to name but a few. Refer to the documentation on the http://docs.oracle.com/javaee/1.4/api/javax/servlet/http/HttpServletResponse.html[HttpServletResponse] class in the Servlet API for more information. Grails provides the following additional methods and/or properties on the `response` object
```
Comment (graemerocher) [inline src/en/ref/Servlet API/response.adoc]: This change too is unnecessary

File: src/en/ref/Servlet API/servletContext.adoc [hunk 18]
```diff
@@ -33,9 +33,9 @@ class BookController {
 === Description
 
 
-The Servlet API's {javaee}javax/servlet/ServletContext.html[ServletContext] is useful for, amongst other things, storing global application attributes, reading local server resources and establishing information about the servlet container.
+The Servlet API's http://docs.oracle.com/javaee/1.4/api/javax/servlet/ServletContext.html[ServletContext] is useful for, amongst other things, storing global application attributes, reading local server resources and establishing information about the servlet container.
```
Comment (graemerocher) [inline src/en/ref/Servlet API/servletContext.adoc]: This change too is unnecessary

File: src/en/ref/Servlet API/servletContext.adoc [hunk 19]
```diff
@@ -6,7 +6,7 @@
 === Purpose
 
 
-The servletContext object is an instance of the Servlet API's {javaee}javax/servlet/ServletContext.html[ServletContext] class.
+The servletContext object is an instance of the Servlet API's http://docs.oracle.com/javaee/1.4/api/javax/servlet/ServletContext.html[ServletContext] class.
```
Comment (graemerocher) [inline src/en/ref/Servlet API/servletContext.adoc]: This change too is unnecessary

File: src/en/ref/Tags/message.adoc [hunk 20]
```diff
@@ -37,7 +37,7 @@ Output a message for a compatible object.
 <g:message message="${myObj}" />
 ----
 
-NOTE: Objects passed to the message parameter must implement the {springapi}org/springframework/context/MessageSourceResolvable.html[MessageSourceResolvable] interface.
+NOTE: Objects passed to the message parameter must implement the http://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/context/MessageSourceResolvable.html[MessageSourceResolvable] interface.
```
Comment (graemerocher) [inline src/en/ref/Tags/message.adoc]: This change too is unnecessary

File: src/en/ref/Tags/fieldValue.adoc [hunk 21]
```diff
@@ -28,3 +28,10 @@ Attributes
 
 * `bean` (required) - The bean instance to inspect
 * `field` (required) - The name of the field to obtain the value of
+
+
+=== Source
+
+
+{source:tag=ValidationTagLib.fieldValue}
+{source}
```
Comment (graemerocher) [inline src/en/ref/Tags/fieldValue.adoc]: This change too is unnecessary

File: src/en/ref/Tags/country.adoc [hunk 22]
```diff
@@ -22,10 +22,17 @@ Supported countries: <g:country code="gbr"/>, <g:country code="deu"/>
...
 Attributes
 
 * `code` - The 3-letter lowercase ISO3166 country code
+
+
+=== Source
+
+
+{source:tag=CountryTagLib.country}
+{source}
```
Comment (graemerocher) [inline src/en/ref/Tags/country.adoc]: This change too is unnecessary

File: src/en/ref/Tag Libraries/session.adoc [hunk 23]
```diff
@@ -30,6 +30,6 @@ class UserController {
 === Description
 
 
-The {javaee}javax/servlet/http/HttpSession.html[HttpSession] class is useful for associated session data with a client.
+The http://docs.oracle.com/javaee/1.4/api/javax/servlet/http/HttpSession.html[HttpSession] class is useful for associated session data with a client.
```
Comment (graemerocher) [inline src/en/ref/Tag Libraries/session.adoc]: This change too is unnecessary

File: src/en/ref/Tag Libraries/session.adoc [hunk 24]
```diff
@@ -6,7 +6,7 @@
 === Purpose
 
 
-The link:../Servlet%20API/session.html[session] object is an instance of the Servlet API's {javaee}javax/servlet/http/HttpSession.html[HttpSession] class
+The link:../Servlet%20API/session.html[session] object is an instance of the Servlet API's http://docs.oracle.com/javaee/1.4/api/javax/servlet/http/HttpSession.html[HttpSession] class
```
Comment (graemerocher) [inline src/en/ref/Tag Libraries/session.adoc]: This change too is unnecessary

File: src/en/ref/Tag Libraries/servletContext.adoc [hunk 25]
```diff
@@ -33,6 +33,6 @@ class BookController {
 === Description
 
 
-The Servlet API's {javaee}javax/servlet/ServletContext.html[ServletContext] is useful for, amongst other things, storing global application attributes, reading local server resources and establishing information about the servlet container.
+The Servlet API's http://docs.oracle.com/javaee/1.4/api/javax/servlet/ServletContext.html[ServletContext] is useful for, amongst other things, storing global application attributes, reading local server resources and establishing information about the servlet container.
```
Comment (graemerocher) [inline src/en/ref/Tag Libraries/servletContext.adoc]: This change too is unnecessary

File: src/en/ref/Tag Libraries/servletContext.adoc [hunk 26]
```diff
@@ -6,7 +6,7 @@
 === Purpose
 
 
-The <link:../Servlet%20API/servletContext.html[servletContext] object is an instance of the Servlet API's {javaee}javax/servlet/ServletContext.html[ServletContext] class.
+The link:../Servlet%20API/servletContext.html[servletContext] object is an instance of the Servlet API's http://docs.oracle.com/javaee/1.4/api/javax/servlet/ServletContext.html[ServletContext] class.
```
Comment (graemerocher) [inline src/en/ref/Tag Libraries/servletContext.adoc]: This change too is unnecessary

File: src/en/ref/Tag Libraries/response.adoc [hunk 27]
```diff
@@ -26,6 +26,6 @@ class BookController {
 === Description
 
 
-The Servlet API's `HttpServletResponse` class can be used within Grails to perform all typical activities such as writing out binary data, writing directly to the response and sending error response codes to name but a few. Refer to the documentation on the {javaee}javax/servlet/http/HttpServletResponse.html[HttpServletResponse] class in the Servlet API for more information.
+The Servlet API's `HttpServletResponse` class can be used within Grails to perform all typical activities such as writing out binary data, writing directly to the response and sending error response codes to name but a few. Refer to the documentation on the http://docs.oracle.com/javaee/1.4/api/javax/servlet/http/HttpServletResponse.html[HttpServletResponse] class in the Servlet API for more information.
```
Comment (graemerocher) [inline src/en/ref/Tag Libraries/response.adoc]: This change too is unnecessary

**Comment by rpalcolea** [comment]
Thanks for the review @graemerocher 

Most of the changes came from master, my bad.

Will fix and notify you as soon as is ready.

Btw, I noticed that master has old info like here: https://github.com/grails/grails-doc/blob/master/src/en/ref/Command%20Line.adoc

It makes reference to Gant instead of Gradle... but branch 3.1.x has the proper instructions

**Comment by rpalcolea** [comment]
Hi @graemerocher 

Rolled back unnecessary changes such as removing `{apiDocs}`, `{javaee}`, tables and content.

Please let me know if you are ok with the changes.
